### PR TITLE
Support multiple C++ ServiceComm target versions

### LINKED ISSUE #52: Support targeting multiple ServiceComm versions in the C++ generator

### PR DESCRIPTION
**Pull request by sigiesec**
... in general, and specifically added support for target ServiceComm 0.11 and 0.12, the latter of which is the new default

Fixes #52 

---
This change is [Reviewable](https://reviewable.io/reviews/btc-ag/service-idl/60)

### CODE REVIEW

**Comment by sigiesec** [comment]
@btc-ag/service-idl Can someone please review this?

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/DefaultGenerationSettingsProvider.xtend [hunk 1]
```diff
@@ -19,20 +19,43 @@ import com.btc.serviceidl.generator.cpp.prins.VSSolutionFactory
...
 
 class DefaultGenerationSettingsProvider implements IGenerationSettingsProvider
 {
-
     public Set<ArtifactNature> languages
     public Set<ProjectType> projectTypes
     public IProjectSetFactory projectSetFactory
     public IModuleStructureStrategy moduleStructureStrategy
+    private static val Map<String, Set<String>> supportedVersionMap = createSupportedVersionMap
+    private val Map<String, String> versionMap
```
Comment (huttenlocher) [inline com.btc.serviceidl/src/com/btc/serviceidl/generator/DefaultGenerationSettingsProvider.xtend]: Just a remark (here and line above): private by default in Xtend (but no problem to declare it explicitly).

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/Main.java [hunk 2]
```diff
@@ -135,13 +159,41 @@ private boolean tryRunGenerator(String[] inputFiles, String outputPath, String p
                 }
             }
         }
+        System.out.println("IDL input is valid.");
+
+        try {
+            configureGenerationSetings(projectSystem, versions);
+        } catch (Exception ex) {
+            System.out.println("Error when configuring generation settings: " + ex);
```
Comment (huttenlocher) [inline com.btc.serviceidl/src/com/btc/serviceidl/generator/Main.java]: Perhaps better use System.err in this case (error)?

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/DefaultGenerationSettingsProvider.xtend (hunk 1 above)
Comment (sigiesec) [reply com.btc.serviceidl/src/com/btc/serviceidl/generator/DefaultGenerationSettingsProvider.xtend]: Yes, I also think it should be left out if it is redundant. I will document some coding conventions.

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/Main.java (hunk 2 above)
Comment (sigiesec) [reply com.btc.serviceidl/src/com/btc/serviceidl/generator/Main.java]: Yes

**Pull request merged**